src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const FROM_SEPARATOR_FORMAT = fromFormat[3];
+  const TO_SEPARATOR_FORMAT = toFormat[3];
+  let year, month, day;
+  const RESULT = [];
+
+  const ARRAY_FROM_DATE = date.split(FROM_SEPARATOR_FORMAT);
+
+  fromFormat.forEach((part, i) => {
+    switch (part) {
+      case 'YYYY':
+        year = ARRAY_FROM_DATE[i];
+        break;
+
+      case 'YY':
+        year = ARRAY_FROM_DATE[i];
+        break;
+
+      case 'MM':
+        month = ARRAY_FROM_DATE[i];
+        break;
+
+      case 'DD':
+        day = ARRAY_FROM_DATE[i];
+        break;
+
+      default:
+        break;
+    }
+  });
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    year = year.slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    year = parseInt(year) < 30 ? '20' + year : '19' + year;
```

Comment:
The conversion logic for 'YYYY' to 'YY' and 'YY' to 'YYYY' could be simplified by incorporating it into the 'forEach' loop on lines 18-39. This would avoid the need for separate 'if' checks and would make the code more concise and easier to follow.